state_machine/plugs/flight/lib/pycubed.py

```diff
@@ -147,289 +147,467 @@ def __init__(self):
+        if sun_sensor_count >= 1:
+            self.hardware['Sun'] = True
 
         # Initialize H-Bridges
+        coils = []
+
         try:
             drv_x = drv8830.DRV8830(self.i2c3, 0x68)  # U6
+            coils.append(drv_x)
+        except Exception as e:
+            print('[ERROR][H-Bridge U6]', e)
+            
+        try:
             drv_y = drv8830.DRV8830(self.i2c3, 0x60)  # U8
+            coils.append(drv_y)
+        except Exception as e:
+            print('[ERROR][H-Bridge U8]', e)
+            
+        try:
             drv_z = drv8830.DRV8830(self.i2c3, 0x62)  # U4
-            coils = [drv_x, drv_y, drv_z]
-            for driver in coils:
-                driver.mode = drv8830.COAST
-                driver.vout = 0
+            coils.append(drv_z)
+        except Exception as e:
+            print('[ERROR][H-Bridge U4]', e)
+            
+        coil_count = 0
+        for driver in coils:
+            driver.mode = drv8830.COAST
+            driver.vout = 0
+            coil_count += 1
+        
+        if coil_count >= 1:
             self.hardware['Coils'] = True
+
+        # Initialize burnwires
+        burnwires = []
+        try:
+            # needed to change pinout from BURN1 to PA15, as BURN1 did not support PWMOut
+            self.burnwire1 = pwmio.PWMOut(microcontroller.pin.PA15, frequency=1000, duty_cycle=0)
+            burnwires.append(self.burnwire1)
         except Exception as e:
-            print('[ERROR][H-Bridges]', e)
+            print('[ERROR][Burn Wire IC1]', e)
+        
+        try:
+            # needed to change pinout from BURN2 to PA18, as BURN2 did not support PWMOut
+            self.burnwire2 = pwmio.PWMOut(microcontroller.pin.PA18, frequency=1000, duty_cycle=0)
+            burnwires.append(self.burnwire2)
+        except Exception as e:
+            print('[ERROR][Burn Wire IC1]', e)
+
+        burnwire_count = 0
+        for burnwire in burnwires:
+            burnwire_count += 1
+        
+        if burnwire_count >= 1:
+            self.hardware['BurnWire'] = True
+        
 
+    # reinit: reinitialize radio, sd, or IMU based upon the contents of string dev
     def reinit(self, dev):
-        dev = dev.lower()
+        # dev is a whitespace-stripped string of all lowercase letters
+        dev = dev.strip().lower()
+
+        # reinitialize device based on string dev
         if dev == 'radio':
-            self.radio2.__init__(self.spi, self._rf_cs, self._rf_rst, self.UHF_FREQ)
+            # should we be reinitializing radio2 or just radio?
+            self.radio.__init__(self.spi, self._rf_cs, self._rf_rst, self.UHF_FREQ)
         elif dev == 'sd':
             self._sd.__init__(self.spi, self._sdcs, baudrate=1000000)
         elif dev == 'imu':
             self.IMU.__init__(self.i2c1)
         else:
-            print('Invalid Device? ->', dev)
+            print('Invalid Device? ->',dev)
+
 
     @property
+    # return the accelerometer reading from the IMU
```

Comment:
For these please use docstrings, rather than comments as this enables better IDE integration.